### PR TITLE
Add support for ti lm3s6965 spi controller driver.

### PR DESCRIPTION
**Pull request by ashokkumar11**

### CODE REVIEW

File: dts/arm/ti/lm3s6965.dtsi [hunk 1]
```diff
@@ -47,6 +47,16 @@
 			status = "disabled";
 			label = "UART_2";
 		};
+
+		spi0: spi@40008000 {
+			compatible = "ti,stellaris-spi";
```
Comment (bravegnu) [inline dts/arm/ti/lm3s6965.dtsi]: "arm,pl022"

File: dts/bindings/spi/ti,stellaris-spi.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#
+# Copyright (c) 2018, Zilogic Systems
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+---
+title: Ti Stellaris
```
Comment (bravegnu) [inline dts/bindings/spi/ti,stellaris-spi.yaml]: Ti -> TI

File: drivers/clock_control/Kconfig.stellaris [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Kconfig - LM3S6965 CCM
+#
+# Copyright (c) 2018, Zilogic Systems.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+menuconfig CLOCK_CONTROL_STELLARIS
```
Comment (bravegnu) [inline drivers/clock_control/Kconfig.stellaris:8]: Is this specific to LM3S6965? If so then it should have a more specific name.

File: drivers/spi/CMakeLists.txt [hunk 4]
```diff
@@ -3,7 +3,7 @@ zephyr_library()
 zephyr_library_sources_ifdef(CONFIG_SPI_DW		spi_dw.c)
 zephyr_library_sources_ifdef(CONFIG_SPI_INTEL		spi_intel.c)
 zephyr_library_sources_ifdef(CONFIG_SPI_STM32		spi_ll_stm32.c)
-zephyr_library_sources_ifdef(CONFIG_SPI_STELLARIS	spi_stellaris.c)
+zephyr_library_sources_ifdef(CONFIG_SPI_STELLARIS	spi_pl022.c)
```
Comment (bravegnu) [inline drivers/spi/CMakeLists.txt]: Should the config macro be CONFIG_SPI_PL022?

File: drivers/spi/spi_pl022.c [hunk 5]
```diff
@@ -16,7 +16,13 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 #include <board.h>
 #include <string.h>
 
-#include "spi_stellaris.h"
+#ifdef CONFIG_SPI_STELLARIS
```
Comment (bravegnu) [inline drivers/spi/spi_pl022.c]: Can we ifdef by CCM driver name instead?

File: dts/arm/ti/lm3s6965.dtsi (hunk 1 above)
Comment (ashokkumar11) [reply dts/arm/ti/lm3s6965.dtsi]: fixed

File: dts/bindings/spi/ti,stellaris-spi.yaml (hunk 2 above)
Comment (ashokkumar11) [reply dts/bindings/spi/ti,stellaris-spi.yaml]: Renamed ti,stellaris-spi.yaml to arm,pl022.yaml, so this changes not required.

**Pull request merged**